src/app/space/create/deployments/apps/collapsible-deployment-info.component.spec.ts

```diff
@@ -0,0 +1,160 @@
+  @Input() chartData: any;
+}
+
+describe('CollapsibleDeploymentInfoComponent', () => {
+  let component: CollapsibleDeploymentInfoComponent;
+  let fixture: ComponentFixture<CollapsibleDeploymentInfoComponent>;
+  let mockSvc: jasmine.SpyObj<DeploymentsService>;
+  let mockEnvironment: Environment;
+
+  beforeEach(() => {
+    mockSvc = createMock(DeploymentsService);
+    mockSvc.getVersion.and.returnValue(Observable.of('1.2.3'));
+    mockSvc.getCpuStat.and.returnValue(Observable.of({ used: 1, quota: 2 }));
+    mockSvc.getMemoryStat.and.returnValue(Observable.of({ used: 3, quota: 4, units: 'GB' }));
+    mockSvc.getAppUrl.and.returnValue(Observable.of('mockAppUrl'));
+    mockSvc.getConsoleUrl.and.returnValue(Observable.of('mockConsoleUrl'));
+    mockSvc.getLogsUrl.and.returnValue(Observable.of('mockLogsUrl'));
+    mockSvc.deleteApplication.and.returnValue(Observable.of('mockDeletedMessage'));
+
+    mockEnvironment = { name: 'mockEnvironment' } as Environment;
+
+    TestBed.configureTestingModule({
+      imports: [CollapseModule.forRoot()],
+      declarations: [
+        CollapsibleDeploymentInfoComponent,
+        FakeDeploymentsDonutComponent,
+        FakeDeploymentGraphLabelComponent,
+        FakePfngChartSparkline
+      ],
+      providers: [
+        { provide: DeploymentsService, useValue: mockSvc }
+      ]
+    });
+
+    fixture = TestBed.createComponent(CollapsibleDeploymentInfoComponent);
+    component = fixture.componentInstance;
+
+    component.collapsed = false;
+    component.applicationId = 'mockAppId';
+    component.environment = mockEnvironment;
+    component.spaceId = 'mockSpaceId';
+
+    fixture.detectChanges();
+  });
+
+  it('should generate unique chartIds for each CollapsibleDeploymentInfoComponent instance', () => {
+    const COLLAPSIBLE_COMPONENT_COUNT = 5;
+    const CHARTS_PER_COLLAPSIBLE_COMPONENT = 2;
+    let uniqueIdSet = new Set();
+    times(COLLAPSIBLE_COMPONENT_COUNT, () => {
+      let collapsibleComponent = new CollapsibleDeploymentInfoComponent(mockSvc);
+      uniqueIdSet.add(collapsibleComponent.cpuConfig.chartId);
+      uniqueIdSet.add(collapsibleComponent.memConfig.chartId);
+    });
+    expect(uniqueIdSet.size).toBe(COLLAPSIBLE_COMPONENT_COUNT * CHARTS_PER_COLLAPSIBLE_COMPONENT);
+  });
+
+  it('should create a child donut component with proper values', () => {
+    let arrayOfComponents = fixture.debugElement.queryAll(By.directive(FakeDeploymentsDonutComponent));
+    expect(arrayOfComponents.length).toEqual(1);
+
+    let container = arrayOfComponents[0].componentInstance;
+    expect(container.applicationId).toEqual('mockAppId');
+  });
+
+  describe('cpu label', () => {
+    let de: DebugElement;
+
+    beforeEach(() => {
+      let charts = fixture.debugElement.queryAll(By.css('.deployment-chart'));
+      let cpuChart = charts[0];
+      de = cpuChart.query(By.directive(FakeDeploymentGraphLabelComponent));
+    });
+
+    it('should use units from service result', () => {
```

Comment:
This test name is inaccurate - the "Cores" unit is just assumed, not specified by a service result. Looks like this is actually wrong in upstream - please fix it while you're here.